src/java/org/apache/cassandra/db/compression/CompressionDictionaryDetailsTabularData.java

```diff
@@ -0,0 +1,304 @@
+     */
+    public static CompositeData fromCompressionDictionaryDataObject(CompressionDictionaryDataObject dataObject)
+    {
+        try
+        {
+            return new CompositeDataSupport(COMPOSITE_TYPE,
+                                            ITEM_NAMES,
+                                            new Object[]
+                                            {
+                                            dataObject.keyspace,
+                                            dataObject.table,
+                                            dataObject.dictId,
+                                            dataObject.dict,
+                                            dataObject.kind,
+                                            dataObject.dictChecksum,
+                                            dataObject.dictLength
+                                            });
+        }
+        catch (OpenDataException e)
+        {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Deserializes data to convenience object to work further with.
+     *
+     * @param compositeData data to create data object from
+     * @return deserialized composite data to convenience object
+     * @throws IllegalArgumentException if values in deserialized object are invalid.
+     * @see CompressionDictionaryDataObject#validate()
+     */
+    public static CompressionDictionaryDataObject fromCompositeData(CompositeData compositeData)
+    {
+        String keyspace = (String) compositeData.get(CompressionDictionaryDetailsTabularData.KEYSPACE_NAME);
+        String table = (String) compositeData.get(CompressionDictionaryDetailsTabularData.TABLE_NAME);
+        long dictId = (Long) compositeData.get(CompressionDictionaryDetailsTabularData.DICT_ID_NAME);
+        byte[] dictionaryBytes = (byte[]) compositeData.get(CompressionDictionaryDetailsTabularData.DICT_NAME);
+        String kind = (String) compositeData.get(CompressionDictionaryDetailsTabularData.KIND_NAME);
+        int checksum = (Integer) compositeData.get(CompressionDictionaryDetailsTabularData.CHECKSUM_NAME);
+        int size = (Integer) compositeData.get(CompressionDictionaryDetailsTabularData.SIZE_NAME);
+
+        CompressionDictionaryDataObject dataObject = new CompressionDictionaryDataObject();
+        dataObject.keyspace = keyspace;
+        dataObject.table = table;
+        dataObject.dictId = dictId;
+        dataObject.dict = dictionaryBytes;
+        dataObject.kind = kind;
+        dataObject.dictChecksum = checksum;
+        dataObject.dictLength = size;
+
+        dataObject.validate();
+
+        return dataObject;
+    }
+
+    public static class CompressionDictionaryDataObject implements Cloneable
+    {
+        public String keyspace;
+        public String table;
+        public long dictId;
+        public byte[] dict;
+        public String kind;
+        public int dictChecksum;
+        public int dictLength;
```

Comment:
nit: add `final` modifier to those public fields. The change would require to add a constructor though

Reply:
This would complicate tests a bit etc. I have made it final but then I used builder to copy it and builder's methods to override any value.

Builder is also validating that object and constructor of this class is made `private` so it can ever be constructed in the code by a builder only.

JSON deserialisation from String to that object works too by adding appropriate annotations on that constructor. We need to deserialize from String (file) to an instance of that object when we go to import.